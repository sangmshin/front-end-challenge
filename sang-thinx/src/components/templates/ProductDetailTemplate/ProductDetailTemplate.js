import React from 'react';
import { object } from 'prop-types';
import idx from 'idx';
import { connect } from 'react-redux';
import { ProductDetailHero, BestSellers } from 'components/sections';
import { getPageData } from 'store/page-data/selectors';

const ProductDetailTemplate = ({ pageData }) => {
  const heroData = idx(pageData, _ => _.hero)
  const bestSellersData = idx(pageData, _ => _.best_sellers)
  if (!pageData || !heroData) return null;

  return (
    <main>
      <ProductDetailHero {...heroData} />
      <BestSellers {...bestSellersData} />
    </main>
  )
};

ProductDetailTemplate.propTypes = {
  pageData: object
}

ProductDetailTemplate.defaultProps = {
  pageData: {}
}

const mapStateToProps = state => ({
  pageData: getPageData(state)
});

export default connect(mapStateToProps)(ProductDetailTemplate);