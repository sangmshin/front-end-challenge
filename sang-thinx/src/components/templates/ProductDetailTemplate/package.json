{
  "name": "ProductDetailTemplate",
  "main": "./ProductDetailTemplate.js"
}
