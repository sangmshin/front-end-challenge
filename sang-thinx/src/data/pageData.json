
{
  "template": "product-detail",
  "hero": {
    "images": [
      {
        "src": "https://thinx-assets.s3.amazonaws.com/highwaist_black_front_01_1024x1024.jpg", "alt": "Hi-Waist front 1"
      },
      {
        "src": "https://thinx-assets.s3.amazonaws.com/highwaist_black_front_02_1024x1024.jpg", "alt": "Hi-Waist front 2"
      },
      {
        "src": "https://thinx-assets.s3.amazonaws.com/highwaist_black_front_03_1024x1024.jpg", "alt": "Hi-Waist front 3"
      },
      {
        "src": "https://thinx-assets.s3.amazonaws.com/highwaist_black_front_04_1024x1024.jpg", "alt": "Hi-Waist front 4"
      },
      {
        "src": "https://thinx-assets.s3.amazonaws.com/highwaist_black_side_1024x1024.jpg", "alt": "Hi-Waist side"
      },
      {
        "src": "https://thinx-assets.s3.amazonaws.com/highwaist_black_back_1024x1024.jpg", "alt": "Hi-Waist back"
      }
    ],
    "product_title": "Hi - Waist",
    "product_droplets": "Heavy days. Holds up to 2 tampon's worth",
    "description": "The phrase \"treat yo'self\" reaches new belly-button brushing heights with the Hi-Waist, and it's here just in time for cuffing season.",
    "price": "35.00",
    "available_colors": [
      "black",
      "beige"
    ],
    "available_sizes": [
      "xsmall",
      "small",
      "medium",
      "large",
      "xlarge",
      "xxlarge"
    ],
    "size_chart": true,
    "savings": {
      "label": "build a cycle set and save up to 20%",
      "url": ""
    }
  },
  "best_sellers": [
    {
      "product_title": "Hiphugger",
      "product_droplets": "heavy days",
      "image": "https://thinx-assets.s3.amazonaws.com/best_seller_1.jpg"
    },
    {
      "product_title": "Cheeky",
      "product_droplets": "light days",
      "image": "https://thinx-assets.s3.amazonaws.com/best_seller_2.jpg"
    },
    {
      "product_title": "Thong",
      "product_droplets": "lightest days",
      "image": "https://thinx-assets.s3.amazonaws.com/best_seller_3.jpg"
    },
    {
      "product_title": "Sport",
      "product_droplets": "medium days",
      "image": "https://thinx-assets.s3.amazonaws.com/best_seller_4.jpg"
    },
    {
      "product_title": "Boyshort",
      "product_droplets": "light days",
      "image": "https://thinx-assets.s3.amazonaws.com/best_seller_5.jpg"
    }
  ]
}
