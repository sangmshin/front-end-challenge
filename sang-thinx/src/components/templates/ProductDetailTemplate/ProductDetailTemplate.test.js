import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { ProductDetailTemplate } from './ProductDetailTemplate';
import pageData from 'data/pageData.json';

const wrapper = shallow(
  <ProductDetailTemplate pageData={pageData} />
);

describe('Component :: ProductDetailTemplate', () => {
  describe('#render', () => {
    it('should match snapshot', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
