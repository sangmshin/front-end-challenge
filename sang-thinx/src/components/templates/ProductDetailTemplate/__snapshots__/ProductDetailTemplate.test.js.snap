// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component :: ProductDetailTemplate #render should match snapshot 1`] = `
<main>
  <Connect(ProductDetailHero)
    available_colors={
      Array [
        "black",
        "beige",
      ]
    }
    available_sizes={
      Array [
        "xsmall",
        "small",
        "medium",
        "large",
        "xlarge",
        "xxlarge",
      ]
    }
    description="The phrase \\"treat yo'self\\" reaches new belly-button brushing heights with the Hi-Waist, and it's here just in time for cuffing season."
    images={
      Array [
        Object {
          "alt": "Hi-Waist front 1",
          "src": "https://thinx-assets.s3.amazonaws.com/highwaist_black_front_01_1024x1024.jpg",
        },
        Object {
          "alt": "Hi-Waist front 2",
          "src": "https://thinx-assets.s3.amazonaws.com/highwaist_black_front_02_1024x1024.jpg",
        },
        Object {
          "alt": "Hi-Waist front 3",
          "src": "https://thinx-assets.s3.amazonaws.com/highwaist_black_front_03_1024x1024.jpg",
        },
        Object {
          "alt": "Hi-Waist front 4",
          "src": "https://thinx-assets.s3.amazonaws.com/highwaist_black_front_04_1024x1024.jpg",
        },
        Object {
          "alt": "Hi-Waist side",
          "src": "https://thinx-assets.s3.amazonaws.com/highwaist_black_side_1024x1024.jpg",
        },
        Object {
          "alt": "Hi-Waist back",
          "src": "https://thinx-assets.s3.amazonaws.com/highwaist_black_back_1024x1024.jpg",
        },
      ]
    }
    price="35.00"
    product_subtitle="Heavy days. Holds up to 2 tampon's worth"
    product_title="Hi - Waist"
    savings={
      Object {
        "label": "build a cycle set and save up to 20%",
        "url": "",
      }
    }
    size_chart={true}
  />
  <BestSellers
    best_selling_products={
      Array [
        Object {
          "image": Object {
            "alt": "Hiphugger",
            "src": "https://thinx-assets.s3.amazonaws.com/best_seller_1.jpg",
          },
          "product_subtitle": "heavy days",
          "product_title": "Shop Hiphugger",
          "url": "",
        },
        Object {
          "image": Object {
            "alt": "Cheeky",
            "src": "https://thinx-assets.s3.amazonaws.com/best_seller_2.jpg",
          },
          "product_subtitle": "light days",
          "product_title": "Shop Cheeky",
          "url": "",
        },
        Object {
          "image": Object {
            "alt": "Thong",
            "src": "https://thinx-assets.s3.amazonaws.com/best_seller_3.jpg",
          },
          "product_subtitle": "lightest days",
          "product_title": "Shop Thong",
          "url": "",
        },
        Object {
          "image": Object {
            "alt": "Sport",
            "src": "https://thinx-assets.s3.amazonaws.com/best_seller_4.jpg",
          },
          "product_subtitle": "medium days",
          "product_title": "Shop Sport",
          "url": "",
        },
        Object {
          "image": Object {
            "alt": "Boyshort",
            "src": "https://thinx-assets.s3.amazonaws.com/best_seller_5.jpg",
          },
          "product_subtitle": "light days",
          "product_title": "Shop Boyshort",
          "url": "",
        },
      ]
    }
  />
</main>
`;
